guiEditor/src/components/propertyTab/propertyGrids/gui/commonControlPropertyGridComponent.tsx

```diff
@@ -66,8 +65,11 @@ export class CommonControlPropertyGridComponent extends React.Component<ICommonC
     }
 
     private _checkAndUpdateValues(propertyName: string, value: string) {
+
+        let vau = (this.props.control as any)["_" +propertyName];
+        let percentage = (vau as ValueAndUnit).isPercentage;

```

Comment:
sounds really unsafe to do without further checks knowing where vau comes from

Reply:
Is all the rest below from 74 to 89 still needed ? sounds redundant with your new code ?

Reply:
The below code it will determine if we switch from % to px or the other way around.

Reply:
coool I ll let you add a comment to make obvious the difference between old and new values.